### PR TITLE
Update fmt to 10.1.0

### PR DESCRIPTION
**Pull request by gracicot**
The fmt library has pushed a breaking change in its minor version 10.1.0.

Cause of the build error with fmt 10.1.0: https://github.com/fmtlib/fmt/commit/8f18e72df576ebfbd088a57a5c6cae8d4d0fc3dc

Even though this code was safe, the changes to improve safety broke `vcpkg::msg::format`. My Solution is to simply store all arguments in `args` in a tuple then unpack it in `fmt::make_format_args` using an index sequence.

This changes makes it compatible with fmt 10.1.0, and upgrades the library.

### CODE REVIEW

File: include/vcpkg/base/messages.h [hunk 1]
```diff
@@ -128,15 +128,33 @@ VCPKG_FORMAT_AS(vcpkg::LocalizedString, vcpkg::StringView);
...
     {
-        return detail::format_message_by_index(m.index, fmt::make_format_args(args.arg()...));
+        return detail::format_impl(std::index_sequence_for<Types...>{}, m, args...);
     }
     template<class... Tags, class... Types>
     void format_to(LocalizedString& s, MessageT<Tags...> m, TagArg<identity_t<Tags>, Types>... args)
     {
-        return detail::format_message_by_index_to(s, m.index, fmt::make_format_args(args.arg()...));
+        return detail::format_to_impl(std::index_sequence_for<Types...>{}, s, m, args...);
     }
```
Comment (BillyONeal) [inline include/vcpkg/base/messages.h:158]: I don't see any reason to involve tuple here. How about:

```
    namespace detail
    {
        template<class... FmtArgs>
        LocalizedString format_impl(size_t index, FmtArgs&&... args)
        {
            // no forward to intentionally make an lvalue here
            return detail::format_message_by_index(index, fmt::make_format_args(args...));
        }

        template<class... FmtArgs>
        void format_to_impl(LocalizedString& s, size_t index, FmtArgs&&... args)
        {
            // no forward to intentionally make an lvalue here
            return detail::format_message_by_index_to(s, index, fmt::make_format_args(args...));
        }
    }

    template<class... Tags, class... Types>
    LocalizedString format(MessageT<Tags...> m, TagArg<identity_t<Tags>, Types>... args)
    {
        return detail::format_impl(m.index, args.arg()...);
    }
    template<class... Tags, class... Types>
    void format_to(LocalizedString& s, MessageT<Tags...> m, TagArg<identity_t<Tags>, Types>... args)
    {
        return detail::format_to_impl(s, m.index, args.arg()...);
    }
```

Comment (gracicot) [reply include/vcpkg/base/messages.h:158]: Oh! Yeah makes sense. Let me change that!

**Comment by gracicot** [comment]
I think the `PR Build / format` check didn't run after my force push

**Comment by gracicot** [comment]
Seems all good now!

**Review by BillyONeal — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the update :)

**Comment by gracicot** [comment]
My pleasure!

**Pull request merged**